### PR TITLE
Make cice build using CMake!

### PR DESCRIPTION
**Pull request by jgfouca**
[BFB]

### CODE REVIEW

File: components/cice/bld/configure [hunk 1]
```diff
@@ -518,7 +518,7 @@ if ($print>=2) { print "CPP definitions set by configure: \'$cfg_cppdefs\'$eol";
 #-----------------------------------------------------------------------------------------------
 
 my $fp_filename      = 'Filepath';             # name of output filepath file
-my $cpp_filename     = 'CICE_cppdefs';         # name of output file for cice's cppdefs
```
Comment (jonbob) [inline components/cice/bld/configure]: Is there a reason we use "CCSM_cppdefs" instead of something with E3SM or CIME?

Comment (jgfouca) [reply components/cice/bld/configure]: It's a legacy name. I just changed it to match the other components.

Comment (jonbob) [reply components/cice/bld/configure]: OK, just curious

**Comment by jgfouca** [comment]
Merged to next.

**Pull request merged**